In [44]:
import pandas as pd
import numpy as np
import pandas_datareader as pdr
import datetime

We are going to retrieve stock prices of the top 10 US companies by their market capitalization.
The list and the corresponding stocks are:
Amazon

Google

facebook

GE

JPM

Exxon mobil

facebook

Berkshare hathaway

johnson & johnson


Microsoft


In [45]:
stock_list=np.array([u'AAPL', u'MSFT', u'XOM', u'AMZN','BRK',u'JNJ', u'JPM','GOOG','GE','FB'])

In [46]:
def get_ticker(tickers,startdate, enddate):
    array_tables=[pdr.get_data_yahoo(e, startdate, enddate) for e in tickers]
    return pd.concat(array_tables, keys=tickers, names=['Ticker', 'Date'])

In [55]:
start=datetime.datetime(2014,1,1)
end=datetime.datetime(2015,1,1)
data=get_ticker(stock_list,startdate, enddate)


In [56]:
data.groupby(level=0).describe()

Open                                                             \
        count        mean        std         min         25%         50%   
Ticker                                                                     
AAPL    250.0   82.378960   9.647206   58.485714   77.213215   83.038574   
MSFT    250.0   29.827680   1.597954   26.379999   29.054999   30.055000   
XOM     250.0   86.477200   3.062434   77.589996   84.785000   86.459999   
AMZN    250.0  220.011040  26.227583  173.809998  192.100002  224.709999   
BRK     251.0    0.115647   0.083305    0.089600    0.099000    0.109300   
JNJ     250.0   66.919200   2.473645   62.040001   64.934999   67.225002   
JPM     250.0   39.171800   3.561581   30.889999   36.202500   39.280001   
GOOG    250.0  320.195071  26.338256  279.123779  300.903389  315.654358   
GE      250.0   20.211440   1.205087   18.080000   19.142499   19.990000   
FB      155.0   24.799613   4.752172   18.080000   20.674999   23.370001   

                                 High                 ...        Adj Close  \
               75%         max  count        mean     ...              75%   
Ticker                                                ...                    
AAPL     88.375358  100.344284  250.0   83.117914     ...        79.475285   
MSFT     30.937501   32.910000  250.0   30.058800     ...        26.839467   
XOM      88.092498   93.570000  250.0   87.023200     ...        75.475628   
AMZN    242.372498  261.739990  250.0  222.679440     ...       243.324997   
BRK       0.113150    0.900000  251.0    0.117086     ...         0.113000   
JNJ      69.005001   72.110001  250.0   67.242200     ...        59.778778   
JPM      41.997498   46.080002  250.0   39.610960     ...        37.073654   
GOOG    338.730347  383.915985  250.0  322.927847     ...       339.208565   
GE       21.015000   23.040001  250.0   20.380920     ...        17.959975   
FB       28.095001   42.049999  155.0   25.300903     ...        28.045000   

                   Volume                                                      \
               max  count          mean           std         min         25%   
Ticker                                                                          
AAPL     90.622932  250.0  1.319642e+08  5.795175e+07  43938300.0  90530825.0   
MSFT     28.298155  250.0  4.793796e+07  1.765974e+07  20842400.0  37125200.0   
XOM      79.840172  250.0  1.500328e+07  6.275483e+06   6385200.0  11682000.0   
AMZN    261.679993  250.0  4.199616e+06  2.472365e+06    984400.0   2746950.0   
BRK       0.900000  251.0  5.026889e+05  5.986902e+05         0.0    148250.0   
JNJ      62.876186  250.0  1.192229e+07  7.343516e+06   5006500.0   8815775.0   
JPM      39.869228  250.0  3.307656e+07  1.994448e+07   8316700.0  22229875.0   
GOOG    382.590942  250.0  5.228172e+06  2.862561e+06   1688000.0   3794050.0   
GE       19.766026  250.0  4.583845e+07  1.808133e+07  15097200.0  35006575.0   
FB       38.230000  155.0  5.464815e+07  5.587757e+07   8108300.0  27999250.0   

                                               
                50%          75%          max  
Ticker                                         
AAPL    121889950.0  156394700.0  376530000.0  
MSFT     45089100.0   54204800.0  165902900.0  
XOM      14072450.0   16551475.0   84747800.0  
AMZN      3667450.0    4899900.0   22116900.0  
BRK        307000.0     627750.0    3739000.0  
JNJ      10297250.0   12622075.0   98440200.0  
JPM      28091350.0   36829200.0  217294200.0  
GOOG      4565850.0    5943025.0   24977900.0  
GE       41931600.0   51350925.0  139352500.0  
FB       42978900.0   65263800.0  573576400.0  

[10 rows x 48 columns]

We notice that for all the  stocks except Facebook, we have 255 entry wich correponds to 255 prices. For facebook we only have 155 entries.

In order not to load data at each execution, we are going to save the dataframe into a csv file.

In [38]:
data.to_csv("data_stocks.csv")

we are going to clean the data, we are going to see if there is any empty cell "NaN" in the dataframe. The command below enable us to see if there is any empty cell.

In [54]:
bool_nan=data.isnull().values.any()
print "is there any emply cell ? ",bool_nan

is there any emply cell ?  False


In [57]:
data.dtypes

Open         float64
High         float64
Low          float64
Close        float64
Adj Close    float64
Volume         int64
dtype: object